features/user/handler/handler.go

```diff
@@ -2,50 +2,136 @@ package handler
 
 import (
 	"net/http"
-	"star-pos/features/user"
+	"star-pos/app/middlewares"
+	userModel "star-pos/features/user/model"
+	"star-pos/features/user/service"
 	"star-pos/utils/response"
 	"strings"
 
 	"github.com/labstack/echo/v4"
 )
 
-type userHandler struct {
-	userService user.ServiceInterface
-}
-
-func New(us user.ServiceInterface) *userHandler {
-	return &userHandler{
-		userService: us,
-	}
-}
-
-func (uh *userHandler) Hello(c echo.Context) error {
+func Hello(c echo.Context) error {
 	return c.JSON(http.StatusOK, response.WebJSONHelloResponse("hello world"))
 }
 
-func (uh *userHandler) CreateAccount(c echo.Context) error {
+func CreateAccount(c echo.Context) error {
 	newRequest := RequestAccount{}
 	errbind := c.Bind(&newRequest)
 	if errbind != nil {
 		return c.JSON(http.StatusInternalServerError, response.WebJSONResponse("error bind data: "+errbind.Error(), nil))
 	}
 
-	newAccountCore := user.UserCore{
-		PhoneNumber:     newRequest.PhoneNumber,
-		Password:        newRequest.Password,
-		PasswordConfirm: newRequest.PasswordConfirm,
+	if newRequest.Password == "" || newRequest.PasswordConfirm == "" {
+		return c.JSON(http.StatusBadRequest, response.WebJSONResponse("this field is required", nil))
+	} else if newRequest.PasswordConfirm != newRequest.Password {
+		return c.JSON(http.StatusBadRequest, response.WebJSONResponse("password confirm didn't match", nil))
 	}
 
-	errCreate := uh.userService.Create(newAccountCore)
+	requestAccount := userModel.User{
+		PhoneNumber: newRequest.PhoneNumber,
+		Password:    newRequest.Password,
+	}
+
+	errCreate := service.Create(requestAccount)
 	if errCreate != nil {
 		if strings.Contains(errCreate.Error(), "phone") {
 			return c.JSON(http.StatusBadRequest, response.WebJSONResponse("error create account: "+errCreate.Error(), nil))
-		} else if strings.Contains(errCreate.Error(), "password") {
-			return c.JSON(http.StatusBadRequest, response.WebJSONResponse("error create account: "+errCreate.Error(), nil))
 		} else {
 			return c.JSON(http.StatusInternalServerError, response.WebJSONResponse("error create account:"+errCreate.Error(), nil))
 		}
 	}
 
 	return c.JSON(http.StatusCreated, response.WebJSONResponse("success create account", nil))
 }
+
+func GetProfile(c echo.Context) error {
+	idToken := middlewares.ExtractTokenUserId(c)
+	result, err := service.GetProfile(idToken)
```

Comment:
please read the task description carefully. /users is to get all user. not current user.